packages/lit-html/src/lib/lit-html.ts

```diff
@@ -229,23 +237,24 @@ class Template {
         }
         lastIndex = regex.lastIndex;
         if (regex === textRegex) {
-          // TODO (justinfagnani): there are more ways to open a comment in
-          // HTML, like `<?` or `</{`. Figure out whether / how to handle that.
           if (match[1] === '!--') {
             regex = commentRegex;
-          } else if (match[2] === undefined) {
-            // If match[2] !== undefined, then we have an unescaped < and stay
-            // in the text state
-            if (rawTextElement.test(match[1])) {
-              regex = new RegExp(`<\/${match[1]}`, 'g');
+          } else if (match[1] !== undefined) {
+            regex = comment2Regex;
+          } else if (match[2] !== undefined) {
+            if (rawTextElement.test(match[2])) {
+              regex = new RegExp(`<\/${match[2]}`, 'g');
             } else {
               regex = tagRegex;
             }
-          } else if (match[2] === '' || match[2] === '/') {
+          } else if (match[3] !== undefined) {
+            // If match[2] !== undefined, then we have an unescaped < and stay
+            // in the text state
```

Comment:
Nit: Depending on how explicit you want to be, we can actually remove the whole 3rd capture group. No need to match  and handle here if we're just going to stay in the same state.